### DISCUSSION TITLE
I need a service enabled after dinit installation and before reboot, while dinitctl doesn't enable services until reboot.

### DISCUSSION THREAD

**Discussion by Folaht** [davmac314/dinit#186] [Q&A] [answered]
Hey there!

I'm trying to build a Artix dinit box using packer. 
There's only one problem: packer needs SSH to be enabled after dinit installation and before the next reboot,  
or it won't be running any further scripts and finish building the box.  

This means I either need dinitctl to be able to run before reboot,  
or have the ssh service enabled without using dinitctl.  

How do I solve this dilemma?

**Comment by davmac314** [comment, marked as answer]
From the manual page for dinitctl:
```
      enable  Persistently enable a waits-for dependency between two services.
              This is much like add-dep but it also starts the dependency if
              the dependent is started (without explicit activation, so the
              dependency will stop if the dependent stops), and it creates a
              symbolic link in the directory specified via the waits-for.d
              directive in the service description (there must be only one
              such directive) so that the dependency will survive between
              sessions.
```

That last bit tells you how you can enable a service for the next boot - create a symbolic link (or any file, actually), with the correct name (`sshd`?) in the `waits-for.d` directory for the boot service (which in Artix is normally `/etc/dinit.d/boot.d`).

**Reply by Folaht** [reply]
That works for artix.  
Unfortunately not for packer, so I'm going to have to figure out what the real issue is with packer.

**Reply by MMOOBB** [reply]
Apologies for necro-post, but wanted to add you may want to enable NetworkManager (symlink it to boot.d) as well. 

Switch over from s6, enabled only sshd, however the server was no longer communicating properly, even with the IP received by DHCP. Fortunately, I was able to connect to the server and enable NetworkManager locally, which solved this issue. 

Just a tip for anyone else finding this in their search engine.
